docs/tutorials/mastering.md

```diff
@@ -960,138 +1106,277 @@ control-plane    traefik-k8s   LoadBalancer  10.152.183.28   10.201.0.53   80:32
-- `upf-gateway`: the IP address of the gateway between the RAN and Access networks
-- `upf-subnet`: subnet where the UPFs are located (also called Access network)
+```console
+cd terraform
+```
+
+Update the `main.tf` file:
 
 ```console
-juju deploy sdcore-gnbsim-k8s gnbsim --channel=beta \
---config gnb-interface=ran \
---config gnb-ip-address=10.204.0.10/24 \
---config icmp-packet-destination=8.8.8.8 \
---config upf-gateway=10.204.0.1 \
---config upf-subnet=10.202.0.0/24
+cat << EOF >> main.tf
+module "gnbsim" {
+  source = "git::https://github.com/canonical/sdcore-gnbsim-k8s-operator//terraform"
+
+  model_name = "gnbsim"
+  
+  config = {
+    gnb-interface           = "ran"
+    gnb-ip-address          = "10.204.0.10/24"
+    icmp-packet-destination = "8.8.8.8"
+    upf-gateway             = "10.204.0.1"
+    upf-subnet              = "10.202.0.0/24"
+  }
+}
+
+resource "juju_integration" "gnbsim-amf" {
+  model = "gnbsim"
+
+  application {
+    name     = module.gnbsim.app_name
+    endpoint = module.gnbsim.fiveg_n2_endpoint
+  }
+
+  application {
+    offer_url = juju_offer.amf-fiveg-n2.url
+  }
+}
+
+resource "juju_offer" "gnbsim-fiveg-gnb-identity" {
+  model            = "gnbsim"
+  application_name = module.gnbsim.app_name
+  endpoint         = module.gnbsim.fiveg_gnb_identity_endpoint
+}
+
+EOF
 ```
 
-Now we use the integration between the gnbsim operator and the AMF operator to exchange the AMF location information.
+Update Juju Terraform provider:
 
 ```console
-juju consume control-plane.amf
-juju integrate gnbsim:fiveg-n2 amf:fiveg-n2
+terraform init
 ```
 
-And present the offering of the gNB identity for integrating with the NMS.
+Deploy SD-Core User Plane:
 
 ```console
-juju offer gnbsim.gnbsim:fiveg_gnb_identity
+terraform apply -auto-approve
 ```
 
+Monitor the status of the deployment:
+
+```console
+watch -n 1 -c juju status --color --relations
+```
+
+The deployment is ready when the `gnbsim` application is in the `Active/Idle` state. 
+It is normal for `grafana-agent` to remain in waiting state.
+
+Log out of the VM.
+
 ## 7. Configure SD-Core
 
-Still on the `juju-controller` VM, switch to the `control-plane` model.
+The following steps show how to configure the SD-Core 5G core network.
+
+We will start by creating integrations between the Network Management System (NMS) and the UPF and the gNB Simulator.
+Once the integrations are ready, we will create the core network configuration: a network slice, a device group and a subscriber.
+
+Log into the `juju-controller` VM:
 
 ```console
-juju switch control-plane
+multipass shell juju-controller
 ```
 
-Relate the UPF to the NMS for exchanging identity information.
+Add required integrations to the `main.tf` file used in the previous steps:
 
 ```console
-juju consume user-plane.upf
-juju integrate upf:fiveg_n4 nms:fiveg_n4
-juju consume gnbsim.gnbsim
-juju integrate gnbsim:fiveg_gnb_identity nms:fiveg_gnb_identity
+cd terraform
```

Comment:
In all other cases like this, the `cd terraform` is a separate block from the `cat`. I have been cheating and simply staying logged into the controller, so I skip the cd commands typically.

We may want to consider keeping the same style here.

Reply:
Done.